ApiBundle/Controller/NodeController.php

```diff
@@ -66,32 +67,48 @@ public function showOrCreateAction(Request $request, $nodeId, $errorNode)
         $currentSiteDefaultLanguage = $currentSiteManager->getCurrentSiteDefaultLanguage();
         $language = $request->get('language', $currentSiteDefaultLanguage);
         $siteId = $currentSiteManager->getCurrentSiteId();
-        $node = $this->findOneNode($nodeId, $language, $siteId, $request->get('version'));
-        if ($node) {
-            $this->denyAccessUnlessGranted($this->getAccessRole($node));
+        $site = $this->get('open_orchestra_model.repository.site')->findOneBySiteId($siteId);
+        $siteGetLanguage = false;
+
+        if (!is_null($site)) {
+            $aliases = $site->getAliases();
+            foreach ($aliases as $alias) {
```

Comment:
@itkg-ngilain I know it. But with this method, whole array will be parsed, create a temporary array add a in_array in each iteration... Look at the code of getLanguages. I don't care about unicity of languages and need first occurence not whole array.